app/services/reporting/agency_and_sp_report.rb

```diff
@@ -48,12 +48,12 @@ def agency_and_sp_emailable_report
     end
 
     def active_agencies
-      @active_agencies ||= Agreements::PartnerAccountStatus.find_by(name: 'active').
-        partner_accounts.
-        includes(:agency).
-        where('became_partner <= ?', report_date).
-        map(&:agency).
-        uniq
+      @active_agencies ||= Agency.joins(:partner_accounts).
+        where(partner_accounts: {
+          partner_account_status: Agreements::PartnerAccountStatus.find_by(name: 'active'),
+          became_partner: ..report_date,
+        }).
+        distinct
```

Comment:
this has a 12x speed improvement

Reply:
I'm wrong here, it's 13,000x faster.

<img width="277" alt="image" src="https://github.com/user-attachments/assets/97f38dc7-cc88-4f27-94c4-2eb27bca444b">